apps/web/src/actions/user-profile-mod.ts

```diff
@@ -12,40 +12,192 @@ import { revalidatePath } from "next/cache";
+				university,
+				major,
+				levelOfStudy,
+				shortID,
+				hackathonsAttended,
+				softwareExperience,
+				heardFrom,
+				shirtSize,
+				dietRestrictions,
+				accommodationNote,
+				GitHub,
+				LinkedIn,
+				PersonalWebsite,
+			})
+			.where(eq(registrationData.clerkID, user.clerkID));
+		return {
+			success: true,
+			newAge: age,
+			newGender: gender,
+			newRace: race,
+			newEthnicity: ethnicity,
+			newWantsToReceiveMLHEmails: wantsToReceiveMLHEmails,
+			newUniversity: university,
+			newMajor: major,
+			newLevelOfStudy: levelOfStudy,
+			newShortID: shortID,
+			newHackathonsAttended: hackathonsAttended,
+			newSoftwareExperience: softwareExperience,
+			newHeardFrom: heardFrom,
+			newShirtSize: shirtSize,
+			newDietaryRestrictions: dietRestrictions,
+			newAccommodationNote: accommodationNote,
+			newGitHub: GitHub,
+			newLinkedIn: LinkedIn,
+			newPersonalWebsite: PersonalWebsite,
+		};
+	},
+);
+
+export const modifyResume = authenticatedAction(
+	z.object({
+		resume: z.string(),
+	}),
+	async ({ resume }, { userId }) => {
+		const user = await db.query.users.findFirst({
+			where: eq(users.clerkID, userId),
+		});
+		if (!user) throw new Error("User not found");
+		await db
+			.update(registrationData)
+			.set({ resume })
+			.where(eq(registrationData.clerkID, user.clerkID));
+		return {
+			success: true,
+			newResume: resume,
+		};
+	},
+);
+
+export const modifyProfileData = authenticatedAction(
+	z.object({
+		pronouns: z.string(),
+		bio: z.string(),
+		skills: z.string().array(),
+		discordUsername: z.string(),
+	}),
+	async ({ pronouns, bio, skills, discordUsername }, { userId }) => {
+		const user = await db
+			.select()
+			.from(users)
+			.leftJoin(profileData, eq(users.hackerTag, profileData.hackerTag))
+			.where(eq(users.clerkID, userId));
+		if (!user || !user[0].profile_data) {
+			throw new Error("User not found");
+		}
 		await db
 			.update(profileData)
-			.set({ bio })
-			.where(eq(profileData.hackerTag, user.hackerTag));
-		return { success: true, newbio: bio };
+			.set({ pronouns, bio, skills, discordUsername })
+			.where(eq(profileData.hackerTag, user[0].profile_data.hackerTag));
+		return {
+			success: true,
+			newPronouns: pronouns,
+			newBio: bio,
+			newSkills: skills,
+			newDiscord: discordUsername,
+		};
 	},
 );
 
 export const modifyAccountSettings = authenticatedAction(
 	z.object({
 		firstName: z.string().min(1).max(50),
 		lastName: z.string().min(1).max(50),
+		//email: z.string().min(1).max(50),
+		hackerTag: z.string().min(1).max(50),
+		hasSearchableProfile: z.boolean(),
 	}),
-	async ({ firstName, lastName }, { userId }) => {
+	async (
+		{ firstName, lastName, hackerTag, hasSearchableProfile },
+		{ userId },
+	) => {
 		const user = await db.query.users.findFirst({
 			where: eq(users.clerkID, userId),
 		});
 		if (!user) throw new Error("User not found");
+		let oldHackerTag = user.hackerTag; // change when hackertag is not PK on profileData table
```

Comment:
note: This is fine for now, but you can also have an `onConflict` statement so the db can handle this for you. With less lookups.